File: FeedItForward/backend/app/schemas/priority_request.py
from pydantic import BaseModel
from typing import Literal
from enum import Enum

from .consumer import Consumer

class PriorityRequestStatusEnum(Enum):
    PENDING = "Pending"
    APPROVED = "Approved"
    REJECTED = "Rejected"

class PriorityRequestHouseCategory(Enum):
    THREE_ROOM = "3 Room"
    FOUR_ROOM = "4 Room"
    FIVE_ROOM = "5 Room"
    EXECUTIVE_CONDO = "Executive Condo"
    CONDO = "Condo"
    LANDED = "Landed"
    PENTHOUSE = "PentHouse"
    OTHERS = "Others"

class PriorityRequest(BaseModel):
    priority_request_id: int
    
    household_income: str
    number_of_residents: int
    occupation: str
    house_category: PriorityRequestHouseCategory
    status: PriorityRequestStatusEnum
    
    date_created: str
    date_updated: str

    consumer_id: int
    consumer: Consumer

class PriorityRequestCreate(BaseModel):
    household_income: str
    number_of_residents: int
    occupation: str
    house_category: PriorityRequestHouseCategory
    
    consumer_id: int

class PriorityRequestUpdate(BaseModel):
    priority_request_id: int
    
    household_income: str
    number_of_residents: int
    occupation: str
    house_category: PriorityRequestHouseCategory
    status: PriorityRequestStatusEnum

File: FeedItForward/frontend/src/schemas/priorityRequest.ts
import { Consumer } from "./consumer";

export const PRIORITY_REQUEST_ACTION = [
  "Approve",
  "Request",
  "Reject",
  "Cancel"
] as const;
export type PriorityRequestActionEnum =
  (typeof PRIORITY_REQUEST_ACTION)[number];

export const PRIORITY_REQUEST_STATUS = [
  "Pending",
  "Approved",
  "Rejected"
] as const;
export type PriorityRequestStatusEnum =
  (typeof PRIORITY_REQUEST_STATUS)[number];

export const PRIORITY_REQUEST_HOUSE_CATEGORY = [
  "3 Room",
  "4 Room",
  "5 Room",
  "Executive Condo",
  "Condo",
  "Landed",
  "PentHouse",
  "Others"
] as const;
export type PriorityRequestHouseCategory =
  (typeof PRIORITY_REQUEST_HOUSE_CATEGORY)[number];

export interface PriorityRequest {
  priority_request_id: number;

  household_income: string;
  number_of_residents: number;
  occupation: string;
  house_category: PriorityRequestHouseCategory;
  status: PriorityRequestStatusEnum;

  date_created: string;
  date_updated: string;

  consumer_id: number;
  consumer: Consumer;
}

export interface PriorityRequestCreate {
  household_income: string;
  number_of_residents: number;
  occupation: string;
  house_category: PriorityRequestHouseCategory;

  consumer_id: number;
}

export interface PriorityRequestUpdate {
  priority_request_id: number;

  household_income: string;
  number_of_residents: number;
  occupation: string;
  house_category: PriorityRequestHouseCategory;
  status: PriorityRequestStatusEnum;
}
